src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1325 @@
+  IL_0009:  ldc.i4.0
+  IL_000a:  blt.s      IL_002a
+  IL_000c:  ldarg.0
+  IL_000d:  ldc.i4.0
+  IL_000e:  ldc.i4.0
+  IL_000f:  newobj     ""System.Index..ctor(int, bool)""
+  IL_0014:  ldc.i4.0
+  IL_0015:  ldc.i4.1
+  IL_0016:  newobj     ""System.Index..ctor(int, bool)""
+  IL_001b:  newobj     ""System.Range..ctor(System.Index, System.Index)""
+  IL_0020:  ldc.i4.5
+  IL_0021:  callvirt   ""int Test2.this[System.Range, int].get""
+  IL_0026:  ldc.i4.1
+  IL_0027:  ceq
+  IL_0029:  ret
+  IL_002a:  ldc.i4.0
+  IL_002b:  ret
+}"),
+                () => verifier.VerifyIL("X.Test3", @"
+{
+  // Code size       48 (0x30)
+  .maxstack  4
+  IL_0000:  ldarg.0
+  IL_0001:  brfalse.s  IL_002e
+  IL_0003:  ldarg.0
+  IL_0004:  callvirt   ""int Test3.Count.get""
+  IL_0009:  ldc.i4.0
+  IL_000a:  blt.s      IL_002e
+  IL_000c:  ldarg.0
+  IL_000d:  ldc.i4.0
+  IL_000e:  ldc.i4.0
+  IL_000f:  newobj     ""System.Index..ctor(int, bool)""
+  IL_0014:  ldc.i4.0
+  IL_0015:  ldc.i4.1
+  IL_0016:  newobj     ""System.Index..ctor(int, bool)""
+  IL_001b:  newobj     ""System.Range..ctor(System.Index, System.Index)""
+  IL_0020:  call       ""int[] System.Array.Empty<int>()""
+  IL_0025:  callvirt   ""int Test3.this[System.Range, params int[]].get""
+  IL_002a:  ldc.i4.1
+  IL_002b:  ceq
+  IL_002d:  ret
+  IL_002e:  ldc.i4.0
+  IL_002f:  ret
+}"),
+                () => verifier.VerifyIL("X.Test4", @"
+{
+  // Code size       56 (0x38)
+  .maxstack  7
+  IL_0000:  ldarg.0
+  IL_0001:  brfalse.s  IL_0036
+  IL_0003:  ldarg.0
+  IL_0004:  callvirt   ""int Test4.Count.get""
+  IL_0009:  ldc.i4.0
+  IL_000a:  blt.s      IL_0036
+  IL_000c:  ldarg.0
+  IL_000d:  ldc.i4.1
+  IL_000e:  newarr     ""System.Range""
+  IL_0013:  dup
+  IL_0014:  ldc.i4.0
+  IL_0015:  ldc.i4.0
+  IL_0016:  ldc.i4.0
+  IL_0017:  newobj     ""System.Index..ctor(int, bool)""
+  IL_001c:  ldc.i4.0
+  IL_001d:  ldc.i4.1
+  IL_001e:  newobj     ""System.Index..ctor(int, bool)""
+  IL_0023:  newobj     ""System.Range..ctor(System.Index, System.Index)""
+  IL_0028:  stelem     ""System.Range""
+  IL_002d:  callvirt   ""int Test4.this[params System.Range[]].get""
+  IL_0032:  ldc.i4.1
+  IL_0033:  ceq
+  IL_0035:  ret
+  IL_0036:  ldc.i4.0
+  IL_0037:  ret
+}"),
+                () => verifier.VerifyIL("X.Test5", @"
+{
+  // Code size       30 (0x1e)
+  .maxstack  4
+  .locals init (int V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  brfalse.s  IL_001c
+  IL_0003:  ldarg.0
+  IL_0004:  callvirt   ""int Test5.Count.get""
+  IL_0009:  stloc.0
+  IL_000a:  ldloc.0
+  IL_000b:  ldc.i4.0
+  IL_000c:  blt.s      IL_001c
+  IL_000e:  ldarg.0
+  IL_000f:  ldc.i4.0
+  IL_0010:  ldloc.0
+  IL_0011:  ldc.i4.0
+  IL_0012:  sub
+  IL_0013:  callvirt   ""int Test5.Slice(int, int)""
+  IL_0018:  ldc.i4.1
+  IL_0019:  ceq
+  IL_001b:  ret
+  IL_001c:  ldc.i4.0
+  IL_001d:  ret
+}")
+            );
+        }
+
+        [Fact]
+        public void ListPattern_UnsupportedTypes()
```

Comment:
Consider keeping tests with types that are not indexable/sliceable because of incorrect signatures:
`this[int x, params object[] other]` // Type is not indexable
`Slice(int x, int y, params object[] other)` // Type is not sliceable